src/guide/loops.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Loops
+---
+
+Loops allow you to repeat an action over and over (and over).
+
+## While Loops
+
+The most basic loop is the `while` loop. The code within the loop repeats until the condition is `false`.
+
+```grain
+let mut i = 0
+
+while (i < 10) {
+  print("looping!")
+  i += 1
+}
+```
+
+Loops can skip to the next iteration using the `continue` keyword. For example, to print only the odd numbers from zero to ten:
+
+```grain
+let mut i = 0
+
+while (i < 10) {
+  if (i % 2 == 0) continue
+  print(i)
+  i += 1
+}
+```
+
+Use the `break` keyword to exit from a loop entirely:
+
+```grain
+let mut i = 0
+
+while (true) {
+  if (i < 10) {
+    print("forever?")
+    i += 1
+  } else {
+    break
+  }
+}
+```
+
+## For Loops
+
+`for` loops are just like `while` loops, just with more structure. `for` loops accept an optional initializer statement which runs before the loop, an optional condition statement which runs before each iteration, and an optional increment statement which runs after each iteration.
+
+`for` loops are commonly used to iterate arrays:
+
+```grain
+import Array from "array"
+
+let strings = [> "foo", "bar", "baz"]
+
+for (let mut i = 0; i < Array.length(strings); i += 1) {
+  print(strings[i])
+}
+```
+
+As all of the loop parameters are optional, they can all be omitted (though the semicolons are still required). Omitting the condition results in an infinite loop:
+
+```grain
+for (;;) {
+  print("forever!")
+}
+```
```

Comment:
It might make sense to tie into `.forEach`, `.map` and `.reduce` here or just mention those over regular loops as they are often more useful it might also make sense to mention recursive looping and tail calls in grain as they are more idiomatic.

Reply:
This should not happen. Those are functions in the standard library and not part of the language or Arrays themselves.

Reply:
I just meant more as referring to the idiomatic way something like 
```
In grain you will commonly use the standard library functions to loop through data types over normal loops
```

Not documenting them just referencing them as an alternative way todo things it can help newer developers discover alternative ways that are better.